src/gosling-mouse-event/polygon.ts

```diff
@@ -12,6 +12,44 @@ export const isPointNearPoint: (point: [number, number], point2: number[], radiu
     return Math.sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2) < radius;
 };
 
+/**
+ * @param {Array} range Tuple of the form `[x1,x2]` to be tested.
+ * @param {Array} point A value x to be tested.
+ * @returns {boolean} If `true` point lies within the point.
+ */
+export const isPointInsideRange: (range: [number, number], x: number) => boolean = ([x1, x2], x) => {
+    return x1 <= x && x <= x2;
+};
+
+/**
+ * @param {Array} range Tuple of the form `[x1,x2]` to be tested.
+ * @param {Array} x A value x to be tested.
+ * @param {number} radius A radius of the point.
+ * @returns {boolean} If `true` point lies within the point.
+ */
+export const isCircleWithinRange: (range: [number, number], x: number, radius?: number) => boolean = (
+    range,
+    x,
+    radius = 5
+) => {
+    return isPointInsideRange(range, x - radius) && isPointInsideRange(range, x + radius);
+};
+
+/**
+ * @param {Array} point Tuple of the form `[x1,x2]` to be tested.
+ * @param {Array} path 1D list of vertices defining the line segments.
+ * @return {boolean} If `true` point lies within the polygon.
+ */
+export const isAnyPointsWithinRange: (range: [number, number], path: number[]) => boolean = ([x1, x2], path) => {
+    let lx;
+    let isWithin = true;
+    for (let i = 0; i < path.length; i += 2) {
+        lx = path[i];
+        isWithin = isWithin && isPointInsideRange([x1, x2], lx);
+    }
+    return isWithin;
+};
```

Comment:
I think this can just early exit, unless I'm missing something

```suggestion
export const isAnyPointsWithinRange: (range: [number, number], path: number[]) => boolean = ([x1, x2], path) => {
    let lx;
    for (let i = 0; i < path.length; i += 2) {
        lx = path[i];
        if (!isPointInsideRange([x1, x2], lx)) {
          return false;
        }
    }
    return true;
};
```

Reply:
Cool, I like `every()`! The `path` is a flat array that contains both x and y values and this array only uses x. I've changed this to

```
path.filter((_, i) => i % 2 === 0).every(x => isPointInsideRange(range, x));
```